homeassistant/components/ebus/__init__.py

```diff
@@ -0,0 +1,467 @@
+"""EBUS Integration."""
+from __future__ import annotations
+
+import asyncio
+import collections
+import copy
+import itertools
+import logging
+
+from pyebus import (
+    NA,
+    OK,
+    CircuitInfo,
+    CircuitMap,
+    CommandError,
+    Ebus,
+    FieldDef,
+    Prioritizer,
+    get_icon,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    API,
+    CHECKINTERVAL,
+    CONF_CIRCUITINFOS,
+    CONF_MSGDEFCODES,
+    DEFAULT_CIRCUITMAP,
+    DEFAULT_PRIO,
+    DOMAIN,
+    PRIO_OFF,
+    PRIO_TIMEDELTAS,
+    SCAN,
+    TTL,
+    UNDO_UPDATE_LISTENER,
+    UNIT_DEVICE_CLASS_MAP,
+)
+
+PLATFORMS = ["sensor"]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set Up The Ebus Component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set Up A Config Entry."""
+    undo_listener = config_entry.add_update_listener(update_listener)
+    api = Api(hass)
```

Comment:
what happens if the ebus-server is unavailable? You should throw a `ConfigEntryNotReady` error in that case.